app/pktgen-stats.c

```diff
@@ -533,14 +534,14 @@ pktgen_process_stats(struct rte_timer *tim __rte_unused, void *arg __rte_unused)
         rate = &info->rate_stats;
         prev = &info->prev_stats;
 
-        rate->ipackets  = curr->ipackets - prev->ipackets;
-        rate->opackets  = curr->opackets - prev->opackets;
-        rate->ibytes    = curr->ibytes - prev->ibytes;
-        rate->obytes    = curr->obytes - prev->obytes;
-        rate->ierrors   = curr->ierrors - prev->ierrors;
-        rate->oerrors   = curr->oerrors - prev->oerrors;
-        rate->imissed   = curr->imissed - prev->imissed;
-        rate->rx_nombuf = curr->rx_nombuf - prev->rx_nombuf;
+        rate->ipackets  = (curr->ipackets - prev->ipackets) * Billion / (double)elapsed_ns + 0.5;
+        rate->opackets  = (curr->opackets - prev->opackets) * Billion / (double)elapsed_ns + 0.5;
+        rate->ibytes    = (curr->ibytes - prev->ibytes) * Billion / (double)elapsed_ns + 0.5;
+        rate->obytes    = (curr->obytes - prev->obytes) * Billion / (double)elapsed_ns + 0.5;
+        rate->ierrors   = (curr->ierrors - prev->ierrors) * Billion / (double)elapsed_ns + 0.5;
+        rate->oerrors   = (curr->oerrors - prev->oerrors) * Billion / (double)elapsed_ns + 0.5;
+        rate->imissed   = (curr->imissed - prev->imissed) * Billion / (double)elapsed_ns + 0.5;
+        rate->rx_nombuf = (curr->rx_nombuf - prev->rx_nombuf) * Billion / (double)elapsed_ns + 0.5;
 
         if (rate->ipackets > 0xffffffff)
```

Comment:
I know these two lines are not part of your changes, I put them in as debug. Could you please remove them for me?